### PR TITLE
Move UniversalAudioService to Jellyfin.Api

### PR DESCRIPTION
**Pull request by Ullmie02**
Only builds once #3775 is merged.
Yes, the `null` parameters are dumb but I can't think of another way.

### CODE REVIEW

File: Jellyfin.Api/Controllers/UniversalAudioController.cs [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+                            null,
+                            null,
+                            null,
+                            null,
+                            mediaSource.TranscodeReasons == null ? null : string.Join(",", mediaSource.TranscodeReasons.Select(i => i.ToString()).ToArray()),
+                            null,
+                            null,
+                            null,
+                            null)
+                        .ConfigureAwait(false);
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/UniversalAudioController.cs:384]:
```suggestion
```

Two bodies are the same

File: Jellyfin.Api/Controllers/UniversalAudioController.cs [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+                        null,
+                        null,
+                        null,
+                        null,
+                        mediaSource.TranscodeReasons == null ? null : string.Join(",", mediaSource.TranscodeReasons.Select(i => i.ToString()).ToArray()),
+                        null,
+                        null,
+                        EncodingContext.Static,
+                        new Dictionary<string, string>())
+                        .ConfigureAwait(false);
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/UniversalAudioController.cs:258]:
```suggestion
```

Identical as outside block

File: Jellyfin.Api/Controllers/UniversalAudioController.cs [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+            [FromQuery] bool? enableRemoteMedia,
+            [FromQuery] bool breakOnNonKeyFrames,
+            [FromQuery] bool enableRedirection = true)
+        {
+            bool isHeadRequest = Request.Method == System.Net.WebRequestMethods.Http.Head;
+            var deviceProfile = GetDeviceProfile(container, transcodingContainer, audioCodec, transcodingProtocol, breakOnNonKeyFrames, transcodingAudioChannels, maxAudioSampleRate, maxAudioBitDepth, maxAudioChannels);
+            _authorizationContext.GetAuthorizationInfo(Request).DeviceId = deviceId;
+
+            var mediaInfoController = new MediaInfoController(_mediaSourceManager, _deviceManager, _libraryManager, _networkManager, _mediaEncoder, _userManager, _authorizationContext, _loggerFactory.CreateLogger<MediaInfoController>(), _serverConfigurationManager);
+            var playbackInfoResult = await mediaInfoController.GetPostedPlaybackInfo(
```
Comment (cvium) [inline Jellyfin.Api/Controllers/UniversalAudioController.cs:155]: I know that you're just trying to migrate 1:1, but at least use DI for this.

Comment (crobibero) [reply Jellyfin.Api/Controllers/UniversalAudioController.cs:155]: Didn't realize that the controllers could be used through DI. I'm going to have a cleanup PR after everything else is merged; I'll add that to the list

Comment (cvium) [reply Jellyfin.Api/Controllers/UniversalAudioController.cs:155]: When an HTTP request comes in it creates a DI Scope that lives throughout the lifetime of that single request. The controllers are part of that life cycle.

**Pull request merged**